python/ql/src/Expressions/Formatting/AdvancedFormatting.qll

```diff
@@ -51,6 +52,7 @@ library class PossibleAdvancedFormatString extends StrConst {
     predicate isExplicitlyNumbered() { exists(this.fieldId(_, _).toInt()) }
 }
```

Comment:
In Line 35 and comments above there are non-matching brackets - is this intended?

Reply:
I don't know for sure, but the notation `[start, end)` does not strike me as obviously wrong. Some people use parentheses to indicate open ends of ranges, and square brackets to indicate closed ends. (So above `start` is included in the range, but `end` is not.) Working with half-open intervals like this is pretty normal when you're talking about substrings of a string. :slightly_smiling_face:

Reply:
Thanks for explaining 😄